File: GUI/dynamic/actions.py
from __future__ import annotations
from dataclasses import dataclass
from typing import Literal


@dataclass
class SetBackground:
    color: str


@dataclass
class SetForeground:
    color: str


@dataclass
class SetTheme:
    name: str  # dark | light | ocean | forest | sunset | cyber


@dataclass
class SetTitle:
    text: str


@dataclass
class ShowNotification:
    message: str
    severity: Literal["information", "warning", "error"] = "information"
    timeout: float = 3.0


@dataclass
class ResetTheme:
    pass


TUIAction = SetBackground | SetForeground | SetTheme | SetTitle | ShowNotification | ResetTheme

File: GUI/frontend/src/dynamic/actions.ts
export type SetBackground    = { action: 'set_background'; color: string };
export type SetForeground    = { action: 'set_foreground'; color: string };
export type SetTheme         = { action: 'set_theme';      name: string };
export type SetTitle         = { action: 'set_title';      text: string };
export type ShowNotification = { action: 'show_notification'; message: string; severity?: string; timeout?: number };
export type ResetTheme       = { action: 'reset_theme' };

export type TUIAction =
  | SetBackground
  | SetForeground
  | SetTheme
  | SetTitle
  | ShowNotification
  | ResetTheme;
